src/services/codefixes/inferFromUsage.ts

```diff
@@ -930,7 +930,19 @@ namespace ts.codefix {
                     low: t => !!(getObjectFlags(t) & ObjectFlags.Anonymous)
                 }];
             let good = removeLowPriorityInferences(inferences, priorities);
-            const anons = good.filter(i => getObjectFlags(i) & ObjectFlags.Anonymous) as AnonymousType[];
+
+            const anons = [];
+            const displayParts: string[] = [];
+            // filter same displayPart type
+            for (const item of good) {
+                const isAnonymousType = getObjectFlags(item) & ObjectFlags.Anonymous;
+                const displayPart = item.checker.typeToString(item);

```

Comment:
I don't like using strings as the key for de-duping. It happens pretty late and means that we're not addressing the cause of the dupes. Why are the dupes happening? Maybe it's possible to construct the types in a way that eliminates them without a de-duping pass.